common/src/main/java/com/wynntils/wynn/model/CharacterManager.java

```diff
@@ -78,12 +82,25 @@ public static void onWorldStateChanged(WorldStateEvent e) {
 
         if (e.getNewState() == WorldStateManager.State.WORLD
                 && e.getOldState() != WorldStateManager.State.CHARACTER_SELECTION) {
-            McUtils.sendMessageToClient(new TextComponent(
-                            "Could not find your class type. Disable auto join (/toggle autojoin) and try again.")
-                    .withStyle(ChatFormatting.DARK_RED));
+            WynntilsMod.info("Scheduling character info query");
+            scanCharacterInfoPage();
         }
     }
 
+    private static void scanCharacterInfoPage() {
+        ScriptedContainerQuery query = ScriptedContainerQuery.builder("Character Info Query")
+                .useItemInHotbar(InventoryUtils.COMPASS_SLOT_NUM)
+                .matchTitle("^§.\\d+§. skill points remaining$")
```

Comment:
matchTitle should support `Pattern`s too.

Reply:
I'm thinking we can expand the API whenever the need arises. Are you ok with that? The "backend" API is very general, we have a "verificator" that gets the container title and menutype (basically what "kind" of container it is), and returns go/no-go if we got the right one. There is currently one implementation that checks for exact string match, and one for regexp (and neither checks menutype), but we can certainly expand this, to one that takes a Pattern for regexp, or whatever.

Reply:
Im fine with this. We can expand if necessary.